### ISSUE TITLE
SC2215: This flag is used as a command name. Bad line break or missing [ .. ]?

### ISSUE THREAD

**Issue by kenorb** [koalaman/shellcheck#1443]
#### For bugs
- Rule Id: SC2162 & SC2215
- My shellcheck version (`shellcheck --version` or "online"): 0.5.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
grep "^[[:alnum:]][0-9A-Za-z_]\+=" "$SET" | while -r read opt; do
  # shellcheck disable=SC2154 # @see: https://github.com/koalaman/shellcheck/issues/1419
  name=${opt%=*}
  eval "$opt"
  value=${!name}
  ! [[ $value == ?(-)+([0-9.]) ]] && value="\"$value\""
  # shellcheck disable=SC1117 # @fixme
  ex "+%s/ $name\s\+=.*;/ $name = $value;/" -scwq "$MQH"
done
```

#### Here's what shellcheck currently says:

```
Line 2:
grep "^[[:alnum:]][0-9A-Za-z_]\+=" "$SET" | while -r read opt; do
                                                  ^-- SC2215: This flag is used as a command name. Bad line break or missing [ .. ]?
```

If I remove `-r`, then I've got:

```
grep "^[[:alnum:]][0-9A-Za-z_]\+=" "$SET" | while read opt; do
                                                  ^-- SC2162: read without -r will mangle backslashes.
```
#### Here's what I wanted or expected to see:

I'm confused what should I do.

**Comment by pawamoy** [comment]
`-r` must be placed after `read`: `while read -r opt`

**Comment by kenorb** [comment]
Thank you, my fault! I was dry fixing by adding `-r`, didn't notice.
